whois-nrtm4/src/main/java/net/ripe/db/nrtm4/RpslObjectEnqueuer.java

```diff
@@ -0,0 +1,127 @@
+package net.ripe.db.nrtm4;
+
+import com.google.common.collect.Lists;
+import net.ripe.db.nrtm4.dao.WhoisObjectRepository;
+import net.ripe.db.nrtm4.domain.WhoisObjectData;
+import net.ripe.db.nrtm4.domain.RpslObjectData;
+import net.ripe.db.nrtm4.domain.SnapshotState;
+import net.ripe.db.whois.common.domain.CIString;
+import net.ripe.db.whois.common.rpsl.AttributeType;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicInteger;
+
+
+@Service
+public class RpslObjectEnqueuer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RpslObjectEnqueuer.class);
+    private static final int BATCH_SIZE = 100;
+    public static final RpslObjectData POISON_PILL = new RpslObjectData(0, 0, null);
+
+    private final String whoisSource;
+    private final WhoisObjectRepository whoisObjectRepository;
+    private Map<CIString, LinkedBlockingQueue<RpslObjectData>> queueMap;
+
+    RpslObjectEnqueuer(
+        @Value("${whois.source}") final String whoisSource,
+        final WhoisObjectRepository whoisObjectRepository
+    ) {
+        this.whoisSource = whoisSource;
+        this.whoisObjectRepository = whoisObjectRepository;
+    }
+
+    RpslObjectQueueRunner getRunner(
+        final SnapshotState snapshotState,
+        final Map<CIString, LinkedBlockingQueue<RpslObjectData>> queueMap
+    ) {
+        return new RpslObjectQueueRunner(
+            whoisObjectRepository,
+            snapshotState,
+            queueMap,
+            CIString.ciString(whoisSource)
+        );
+    }
+
+    private static class RpslObjectQueueRunner implements Runnable {
```

Comment:
This could be a record, right?

Reply:
yes -- i just noticed the notice from intellij. but do you think it's a good idea to have a 'runnable record' class? there are no getters for these constructor vars and i don't think there should be any, so it's not /really/ a record, nor it used like one (i.e. it implements functionality, not a data class like a pojo).

Reply:
so we only will use record for pojo.
ok!